File: RedSheet.DbEntities/Models/vLiteMeetingManagementTimingRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vLiteMeetingManagementTimingRecords", Schema = "dbo")]
	[Serializable]
	public partial class vLiteMeetingManagementTimingRecord  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int LiteMeetingManagementId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int LiteMeetingManagementTimingId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NegotiationPhaseId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Process { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int SortOrder { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int TacticId { get; set; } 
 
	[Required]
	public TimeSpan Time { get; set; } 

	public vLiteMeetingManagementTimingRecord()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/v-lite-meeting-management-timing-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vLiteMeetingManagementTimingRecord {
    constructor(vLiteMeetingManagementTimingRecord?: vLiteMeetingManagementTimingRecord )  {
        let properties = [ "liteMeetingManagementId", "liteMeetingManagementTimingId", "negotiationPhaseId", "process", "sortOrder", "tacticId", "time",];
        for (let property of properties)
            if (vLiteMeetingManagementTimingRecord && vLiteMeetingManagementTimingRecord[property])
                this[property] = vLiteMeetingManagementTimingRecord[property];
    }
 
	liteMeetingManagementId : number =   0 ;
 
    @range(1,2147483647)
	liteMeetingManagementTimingId : number =   undefined;
 
    @range(1,2147483647)
	negotiationPhaseId : number =   undefined;
 
    @required()
    @maxLength(1000)
	process : string =   undefined;
 
    @range(1,2147483647)
	sortOrder : number =   undefined;
 
    @range(1,2147483647)
	tacticId : number =   undefined;
 
    @required()
	time : Date =   undefined;


}
